src/test/java/org/springframework/data/elasticsearch/core/query/CriteriaQueryTests.java

```diff
@@ -777,7 +778,7 @@ public void shouldPerformBoostOperation() {
 		CriteriaQuery criteriaQuery = new CriteriaQuery(new Criteria("message").contains("foo").boost(1));
 
 		// when
-		Page<SampleEntity> page = operations.queryForPage(criteriaQuery, SampleEntity.class, index);
+		Page<SearchHit<SampleEntity>> page = operations.queryForPage(criteriaQuery, SampleEntity.class, index);
```

Comment:
Would it also make sense to introduce a `SearchPage` type to avoid the deep generic parametrization?